tests/test_BosTau.py

```diff
@@ -32,83 +33,127 @@ def test_basic_attributes(self):
+            "11": 106982474,
+            "12": 87216183,
+            "13": 83472345,
+            "14": 82403003,
+            "15": 85007780,
+            "16": 81013979,
+            "17": 73167244,
+            "18": 65820629,
+            "19": 63449741,
+            "20": 71974595,
+            "21": 69862954,
+            "22": 60773035,
+            "23": 52498615,
+            "24": 62317253,
+            "25": 42350435,
+            "26": 51992305,
+            "27": 45612108,
+            "28": 45940150,
+            "29": 51098607,
+            "X": 139009144,
+            "MT": 16338,
+        }.items(),
+    )
+    def test_chromosome_lengths(self, name, length):
+        assert length == self.genome.get_chromosome(name).length
+
+    # Performs the calculation described in the species file comments
+    lengths = [c.length for c in genome.chromosomes]
+    total_length = sum(lengths) - genome.get_chromosome("MT").length
+    avg_crossovers = (25.5 + 23.2) / 2  # average of male and female numbers
+    avg_rate = avg_crossovers / total_length
+
     @pytest.mark.parametrize(
         ["name", "rate"],
         {
-            "1": -1,
-            "2": -1,
-            "3": -1,
-            "4": -1,
-            "5": -1,
-            "6": -1,
-            "7": -1,
-            "8": -1,
-            "9": -1,
-            "10": -1,
-            "11": -1,
-            "12": -1,
-            "13": -1,
-            "14": -1,
-            "15": -1,
-            "16": -1,
-            "17": -1,
-            "18": -1,
-            "19": -1,
-            "20": -1,
-            "21": -1,
-            "22": -1,
-            "23": -1,
-            "24": -1,
-            "25": -1,
-            "26": -1,
-            "27": -1,
-            "28": -1,
-            "29": -1,
-            "X": -1,
-            "MT": -1,
+            "1": avg_rate,
+            "2": avg_rate,
+            "3": avg_rate,
+            "4": avg_rate,
+            "5": avg_rate,
+            "6": avg_rate,
+            "7": avg_rate,
+            "8": avg_rate,
+            "9": avg_rate,
+            "10": avg_rate,
+            "11": avg_rate,
+            "12": avg_rate,
+            "13": avg_rate,
+            "14": avg_rate,
+            "15": avg_rate,
+            "16": avg_rate,
+            "17": avg_rate,
+            "18": avg_rate,
+            "19": avg_rate,
+            "20": avg_rate,
+            "21": avg_rate,
+            "22": avg_rate,
+            "23": avg_rate,
+            "24": avg_rate,
+            "25": avg_rate,
+            "26": avg_rate,
+            "27": avg_rate,
+            "28": avg_rate,
+            "29": avg_rate,
+            "X": avg_rate,
+            "MT": avg_rate,
         }.items(),
     )
     def test_recombination_rate(self, name, rate):
         assert rate == pytest.approx(
-            self.genome.get_chromosome(name).recombination_rate
+            self.genome.get_chromosome(name).recombination_rate, abs=0.001
```

Comment:
I think as written this will always pass, because recombination rates are like 1e-8, so anything is equal to within +/- 0.001.
```suggestion
            self.genome.get_chromosome(name).recombination_rate, rel=0.001
```